scripts/doc_gen.py

```diff
@@ -0,0 +1,375 @@
+        section_body = self.replace_blank_lines(self.doc.description)
+        return section + section_body + "\n"
+
+    def get_options(self):
+        if not self.doc.options:
+            return
+        section = self.format_section("OPTIONS")
+        section_body = []
+        for option, description in self.doc.options:
+            section_body.append(self.doc.INDENT)
+            parts = option.split(maxsplit=1)
+            name, desc = parts if len(parts) == 2 else (parts[0], "")
+            section_body.append(f"\t{name} {desc}")
+            section_body.append(f"\t{self.replace_blank_lines(description)}")
+        return section + "\n".join(section_body) + "\n"
+
+    def get_commands(self):
+        if not self.doc.commands:
+            return
+        section = self.format_section("COMMANDS")
+        section_body = []
+        for name, description in self.doc.commands:
+            section_body.append(f"\t{self.doc.PARAGRAPH}")
+            section_body.append(f"\t{name}")
+            section_body.append(f"\t{self.replace_blank_lines(description)}")
+            section_body.append(
+                f"\tSee {self.doc.command}-{name}(1) for full documentation on the {name} command."
+            )
+        return section + "\n".join(section_body) + "\n" if section_body else None
+
+    def get_footer(self):
+        footer = f"{self.doc.version}   {self.doc.date}   {self.doc.command}"
+        return footer + "\n"
+
+    def write(self, target_dir: str = None):
+        title = self.get_title()
+        name = self.get_name()
+        synopsis = self.get_synopsis()
+        description = self.get_description()
+        options = self.get_options()
+        commands = self.get_commands()
+        footer = self.get_footer()
+        doc = getattr(self.doc, "doc", None)
+        text_page = "\n".join(
+            filter(
+                None,
+                [
+                    title,
+                    name,
+                    synopsis,
+                    description,
+                    options,
+                    commands,
+                    doc,
+                    footer,
+                ],
+            )
+        )
+        help_page = Path(f'{self.doc.command.replace(" ", "-")}')
+        if target_dir:
+            help_page = Path(target_dir) / help_page
+        help_page.write_text(text_page)
+
+
+def main():
+    if len(sys.argv) < 3:
+        print(f"USAGE: {sys.argv[0]} INPUT_DIR OUTPUT_DIR")
+        sys.exit(2)
+
+    name = "kart"
+    input_dir = sys.argv[1]
+    output_dir = Path(sys.argv[2])
+    output_dir.mkdir(parents=True, exist_ok=True)
+    entry_point = load_entry_point("kart", "console_scripts", "kart")
+    ep_module = import_module(entry_point.value.split(":")[0])
+    ep_module._load_all_commands()
+    ep_members = getmembers(ep_module, lambda x: isinstance(x, click.Command))
+    _, cli = ep_members[0]
+
+    generate_help_pages(name, input_dir, cli, target_dir=output_dir)
```

Comment:
I'm wondering what the advantage is of using the entrypoints mechanism here. Why not just 

```suggestion
    from kart import cli
    cli._load_all_commands()
    generate_help_pages(name, input_dir, cli.cli, target_dir=output_dir)
```

I'd suggest removing the leading underscore from `_load_all_commands` since it's called outside the module.

Reply:
Certainly much better. Thanks :+1: